pkg/provider/kubernetes/crd/traefikio/v1alpha1/middleware.go

```diff
@@ -83,12 +84,18 @@ type CircuitBreaker struct {
 	// Expression is the condition that triggers the tripped state.
 	Expression string `json:"expression,omitempty" toml:"expression,omitempty" yaml:"expression,omitempty" export:"true"`
 	// CheckPeriod is the interval between successive checks of the circuit breaker condition (when in standby state).
+	// +kubebuilder:validation:Pattern="^[0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h)?$"
+	// +kubebuilder:validation:XIntOrString
```

Comment:
```suggestion
```

Isn't it redundant with the regex pattern?

Reply:
According to the kubebuilder doc this is needed to apply patterns:
> This is required when applying patterns or other validations to an IntOrString field. Known information about the type is applied during the collapse phase and as such is not normally available during marker application.

So nevermind my previous comment.